builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -630,46 +720,84 @@ func resourceVSphereVirtualMachineCreate(d *schema.ResourceData, meta interface{
 	}
 
 	if vL, ok := d.GetOk("disk"); ok {
-		disks := make([]hardDisk, len(vL.([]interface{})))
-		for i, v := range vL.([]interface{}) {
-			disk := v.(map[string]interface{})
-			if i == 0 {
+		if diskSet, ok := vL.(*schema.Set); ok {
+
+			disks := []hardDisk{}
+			hasBootableDisk := false
+			for _, value := range diskSet.List() {
```

Comment:
Since the disk schema attribute is set to Required: true, we shouldn't be able to get to that scenario; however, looking at the set.List() implementation, it looks safe for empty sets.